source/MapPanel.cpp

```diff
@@ -1173,28 +1192,39 @@ void MapPanel::DrawTooltips()
 	// Create the tooltip text.
 	if(tooltip.empty())
 	{
-		pair<pair<int, int>, int> t = escortSystems.at(hoverSystem);
+		MapPanel::SystemTooltipData t = escortSystems.at(hoverSystem);
+
 		if(hoverSystem == &playerSystem)
 		{
-			--t.first.first;
-			if(t.first.first || t.first.second || t.second)
+			if(player.Flagship())
+				--t.activeShips;
+			if(t.activeShips || t.parkedShips || t.outfits.size())
 				tooltip = "You are here, with:\n";
 			else
 				tooltip = "You are here.";
 		}
 		// If you have both active and parked escorts, call the active ones
 		// "active escorts." Otherwise, just call them "escorts."
-		if(t.first.first && t.first.second)
-			tooltip += to_string(t.first.first) + (t.first.first == 1 ? " active escort\n" : " active escorts\n");
-		else if(t.first.first)
-			tooltip += to_string(t.first.first) + (t.first.first == 1 ? " escort" : " escorts");
-		if(t.first.second)
-			tooltip += to_string(t.first.second) + (t.first.second == 1 ? " parked escort" : " parked escorts");
-		if(t.second)
+		if(t.activeShips && t.parkedShips)
+			tooltip += to_string(t.activeShips) + (t.activeShips == 1 ? " active escort\n" : " active escorts\n");
+		else if(t.activeShips)
+			tooltip += to_string(t.activeShips) + (t.activeShips == 1 ? " escort" : " escorts");
+		if(t.parkedShips)
+			tooltip += to_string(t.parkedShips) + (t.parkedShips == 1 ? " parked escort" : " parked escorts");
+		if(t.outfits.size())
 		{
-			if(t.first.first || t.first.second)
+			if(t.activeShips || t.parkedShips)
 				tooltip += "\n";
-			tooltip += to_string(t.second) + (t.second == 1 ? " stored outfit" : " stored outfits");
+
+			unsigned sum = 0;
+			for(const auto& it : t.outfits)
+				sum += it.second;
+
+			tooltip += to_string(sum) + (sum == 1 ? " stored outfit" : " stored outfits");
+
+			if(HasMultipleLandablePlanets(hoverSystem) || t.outfits.size() > 1)
+				for(const auto& it : t.outfits)
+					tooltip += "\n - " + to_string(it.second) + " on " + it.first->Name();
```

Comment:
This is pretty much the only part of the PR I don't particularly like, since it is putting the planet-specific data onto the system tooltip.

That said, it won't cause any crazy-looking tooltips in vanilla, since it requires X outfitters in the system to get X new lines (and we have much ghastlier UI issues with 3+ _landable_ objects).

I'm fine with this as a stopgap, but do want to see the per-planet system-level callout get removed in the follow up that augments the orbits UI.

Reply:
When this becomes an issue, then we can always change the tooltip to "You have <nr> outfits stored over <nr> planets in this system."
The issue is for now not very likely, because it will only be an issue when there are many outfitters in a system, and when the player decides to store outfits on many places.